python/pyiceberg/table/__init__.py

```diff
@@ -415,3 +416,8 @@ def to_duckdb(self, table_name: str, connection: Optional[DuckDBPyConnection] =
         con.register(table_name, self.to_arrow())
 
         return con
+
+    def to_ray(self) -> ray.data.dataset.Dataset:
+        import ray
+
+        return ray.data.from_arrow(self.to_arrow())
```

Comment:
just for my understanding, would `self.to_arrow()` returns a `pa.Table` has all rows from Iceberg table? So that would assume the table is very small to fit in the memory of Python client, right? What if the table is larger than the available memory?

Reply:
Thank you for mentioning this. `to_arrow()` will convert all the rows and fields(columns) that are included in the current table scan to a `pa.Table`. Row filters or field selectors could be applied to the table scan to exclude unwanted rows or fields. e.g.
```python
 not_nan = table_test_null_nan.scan(row_filter=NotNaN("col_numeric"), selected_fields=("idx",)).to_arrow()
```

The huge table size will be an issue for `to_arrow` if the table scan ends up including the whole table (or too many rows). There is an open PR #7163 that may help in this case by adding a limit to the number of rows included in the table scan. Here is one example quoted from that PR:
```python
limited_result = table_test_limit.scan(selected_fields=("idx",), limit=20).to_arrow()
```

In the future, I think there could be more discussions on converting large iceberg table to ray dataset. Please let me know if you have more questions/concerns on this.

Reply:
Is this the same experience today for connections like duckdb?

Reply:
I think so. I draw the above conclusion based on the discussion here: https://github.com/apache/iceberg/issues/6505#issuecomment-1453357137, https://github.com/apache/iceberg/issues/6505#issuecomment-1453505015

An out-of-memory error occurs when converting a large table to duckdb on a machine with insufficient memory.

Reply:
Got it. I guess we need to address this issue across connections then. Thanks for the explanation!